### ISSUE TITLE
[Documentation] ServiceMode typo in Custom Service Example Documentation

### ISSUE THREAD

**Issue by cbevans** [coreemu/core#887]
**Describe the bug**
In the Custom Services section of the documentation, it describes the ServiceMode options as blocking, non-blocking, and timer. These do not match the ServiceMode enum of BLOCKING, NON_BLOCKING, and TIMER.  These are case sensitive, and the NON_BLOCKING should be an underscore, not a dash.

The following line in the services.md doc file should be updated:
# validation mode, blocking, non-blocking, and timer
to
# validation mode, BLOCKING, NON_BLOCKING, and TIMER

**Screenshots**
<img width="462" alt="Screenshot 2024-08-01 174341" src="https://github.com/user-attachments/assets/cf725019-62e8-4eda-9f11-df2bf712b6e0">

**Additional context**
FIle: https://github.com/coreemu/core/blob/master/docs/services.md

**Comment by bharnden** [comment]
That is just a comment talking to the modes and was not meant to be literal. If it causes confusion, could look to change to help avoid though.

**Comment by cbevans** [comment]
I took it to be literal - but I'm a new user to it,. I figured out the case sensitivity quickly - but it was the NON-BLOCKING vs. NON_BLOCKING (dash vs. underscore) and I couldn't understand why NON-BLOCKING wasn't working because that was listed in the comment right above. It took digging in the source code to figure it out and it could have been avoided by listing the supported modes right above the line in question.
